### DISCUSSION TITLE
milvus sdk of Java,  does MilvusServiceClient class thread safe ?

### DISCUSSION THREAD

**Discussion by YangZhaoo** [milvus-io/milvus#29001] [Q&A and General discussion] [answered]
I used MilvusServiceClient in my application as a singleton object, I want to know whether it's thread safe?
thank you

**Comment by yhmo** [comment, marked as answer]
All the rpc interfaces are thread-safe because the underlying grpc classes are thread-safe. But some configure methods such as withTimeout/witRetry/close are not guaranteed to be thread-safe.
